File: ios/AnnotationAccPackKit/OTAnnotationAccPackKit/SendAnnotationViewController.m
//
//  RemoteAnnotationViewController.m
//  OTAnnotationAccPackKit
//
//  Created by Xi Huang on 7/18/16.
//  Copyright © 2016 Tokbox, Inc. All rights reserved.
//

#import "SendAnnotationViewController.h"
#import <OTAnnotationKit/OTAnnotationKit.h>
#import <OTScreenShareKit/OTScreenShareKit.h>

@interface SendAnnotationViewController () <AnnotationDelegate>
@property (nonatomic) OTAnnotator *annotator;
@property (nonatomic) OTScreenSharer *sharer;
@end

@implementation SendAnnotationViewController

- (void)viewDidLoad {
    [super viewDidLoad];

    /**
     *  UNCOMMENT THIS TO START TESTING
     */
    self.sharer = [OTScreenSharer screenSharer];
    [self.sharer connectWithView:[UIApplication sharedApplication].keyWindow.rootViewController.view
                         handler:^(ScreenShareSignal signal, NSError *error) {
                             
                             if (!error) {
                                 NSLog(@"Start screen sharing");
                             }
                         }];
    
    self.annotator = [OTAnnotator annotator];
    self.annotator.delegate = self;
    [self.annotator connectForSendingAnnotation];
    self.annotator.annotationView.currentAnnotatable = [OTAnnotationPath pathWithStrokeColor:[UIColor yellowColor]];
}

- (void)annotationWithSignal:(OTAnnotationSignal)signal
                       error:(NSError *)error {
    
    if (signal == OTAnnotationSessionDidConnect){
        [self.view addSubview:self.annotator.annotationView];
    }
}

@end

File: ios/AnnotationAccPackKit/OTAnnotationAccPackKit/SendAnnotationViewController.swift
//
//  SendAnnotationViewController.swift
//  OTAnnotationAccPackKit
//
//  Created by Xi Huang on 9/10/16.
//  Copyright © 2016 Tokbox, Inc. All rights reserved.
//

import Foundation

class SendAnnotationViewController: UIViewController, AnnotationDelegate {
    
    let annotator = OTAnnotator()
    let sharer = OTScreenSharer.sharedInstance()
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        navigationController?.navigationBar.translucent = false
        sharer.connectWithView(nil) {
            [unowned self]
            (signal: ScreenShareSignal, error: NSError!) in
            
            if error == nil {
                
                if signal == .SessionDidConnect {
                    self.sharer.publishAudio = false
                    self.sharer.publishVideo = false
                }
                else if signal == .SubscriberConnect {
                    
                    self.sharer.subscriberView.removeFromSuperview()
                    self.sharer.subscriberView.frame = self.view.bounds
                    self.view.insertSubview(self.sharer.subscriberView, atIndex: 0)
                }
            }
        }
        
        annotator.delegate = self
        annotator.connectForSendingAnnotation()
        annotator.annotationView.currentAnnotatable = OTAnnotationPath.init(strokeColor: UIColor.yellowColor())
    }
    
    func annotationWithSignal(signal: OTAnnotationSignal, error: NSError!) {
        
        if signal == .SessionDidConnect {
            annotator.annotationView.frame = view.frame
            view.addSubview(annotator.annotationView)
        }
    }
}
